gender_novels/corpus.py

```diff
@@ -256,6 +256,83 @@ def get_wordcount_counter(self):
             corpus_counter += novel_counter
         return corpus_counter
 
+
+    def get_novel(self, metadata_field, field_val):
+        """
+        Returns a specific Novel object from self.novels that has metadata matching field_val for
+        metadata_field.  Otherwise raises a ValueError.
+        N.B. This function will only return the first novel in the self.novels (which is sorted as
+        defined by the Novel.__lt__ function).  It should only be used if you're certain there is
+        only one match in the Corpus or if you're not picky about which Novel you get.  If you want
+        more selectivity use get_novel_multiple_fields, or if you want multiple novels use the subcorpus
+        function.
+
+        >>> from gender_novels.corpus import Corpus
+        >>> c = Corpus('sample_novels')
+        >>> c.get_novel("author", "Dickens, Charles")
+        <Novel (dickens_twocities)>
+        >>> c.get_novel("date", '1857')
+        <Novel (bronte_professor)>
+        >>> try:
```

Comment:
There is an easier way of checking for an error:
```
         >>> c.get_novel("meme_quality", "over 9000")
         Traceback (most recent call last):
         ...
         AttributeError: Metadata field meme_quality invalid for this corpus
```

Reply:
Doesn't work on my machine for some reason.

Reply:
Interesting. Can you post the error you're getting?

Reply:
Testing started at 12:20 PM ...
/usr/bin/python3.6 /opt/pycharm-2018.2.4/helpers/pycharm/docrunner.py /home/fsae/mingfei_temp/gender_novels/gender_novels/corpus.py::Corpus::get_novel

Error
**********************************************************************
File "/home/fsae/mingfei_temp/gender_novels/gender_novels/corpus.py", line 276, in get_novel
Failed example:
    c.get_novel("meme_quality", "over 9000") # doctest:+ELLIPSIS
Exception raised:
    Traceback (most recent call last):
      File "/opt/pycharm-2018.2.4/helpers/pycharm/docrunner.py", line 140, in __run
        compileflags, 1), test.globs)
      File "<doctest get_novel[4]>", line 1, in <module>
        c.get_novel("meme_quality", "over 9000") # doctest:+ELLIPSIS
      File "/home/fsae/mingfei_temp/gender_novels/gender_novels/corpus.py", line 287, in get_novel
        raise AttributeError(f"Metadata field {metadata_field} invalid for this corpus")
    AttributeError: Metadata field meme_quality invalid for this corpus


Process finished with exit code 0

Reply:
It doesn't recognize the error as part of the doctest, it seems.

Reply:
Curious. But the test you have works so we can leave it for the moment. Can you resolve the merge conflicts on your end (I made changes to corpus and corpus_gen) and then push you're code again? I'll then merge it into upstream.